tower-grpc/src/generic/codec.rs

```diff
@@ -81,7 +81,7 @@ enum EncodeInner<T, U> {
         /// The source of messages to encode
         inner: U,
     },
-    Err(Status),
+    Err,
```

Comment:
Does this now remove the ability to return a trailing status code when the inner stream bails out?

Reply:
With this change, if an error occurs on the inner stream `status` is now included in the headers and `Encode::is_end_stream` would return `true`.

If the inner stream does not initially error, but bails out a later point, would it be unable to send the `status` back to the client because it is now part of the headers? If that is the change in behavior that was not intended!

This starts to get into the strategy used to solve this, so I'll respond to your comment down below.